android/app/src/main/java/org/smartregister/fhircore/auth/account/AccountHelper.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.auth.account
+
+import android.accounts.Account
+import android.accounts.AccountManager
+import android.accounts.AccountManagerCallback
+import android.accounts.NetworkErrorException
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import android.os.Handler
+import io.jsonwebtoken.Jwts
+import java.util.Date
+import okhttp3.ResponseBody
+import org.smartregister.fhircore.BuildConfig
+import org.smartregister.fhircore.api.OAuthService
+import org.smartregister.fhircore.auth.OAuthResponse
+import retrofit2.Call
+import retrofit2.HttpException
+import retrofit2.Response
+import timber.log.Timber
+
+class AccountHelper(context: Context) {
+  private val mContext = context
+
+  fun getUserInfo(): Call<ResponseBody> {
+    return OAuthService.create(mContext).userInfo()
+  }
+
+  fun refreshToken(refreshToken: String): OAuthResponse? {
+    val data = buildOAuthPayload("refresh_token")
+
+    data["refresh_token"] = refreshToken
+
+    return OAuthService.create(mContext).fetchToken(data).execute().body()
+  }
+
+  @Throws(NetworkErrorException::class)
+  fun fetchToken(username: String, password: CharArray): Call<OAuthResponse> {
+    val data = buildOAuthPayload("password")
+
+    data["username"] = username
+    data["password"] = password.concatToString()
```

Comment:
Is it possible to only use Char's for passwords?

Reply:
cc @maimoonak

Reply:
The API accepts string password. Changed it in other places though